### PR TITLE
Bump `vertx`, `jackson`, `netty` and a couple of other deps

### PR DESCRIPTION
**Pull request by gaetanmaisse**
**Issue**

NA

**Description**

Versions for `jackson` and `netty` are the ones defined in `vertx-dependencies`

BREAKING CHANGE: Vert.x update might contain breaking changes.

Other dependencies will be updated in another PR

---
**Gravitee.io Automatic Deployment**

🚀 A prerelease version of this package has been published on Gravitee's private artifactory, you can:
 - use it directly by updating your project with version: `4.0.0-dep-major-update-SNAPSHOT`
 - download it from Artifactory [here](https://odbxikk7vo-artifactory.services.clever-cloud.com/gravitee-snapshots/io/gravitee/gravitee-bom/4.0.0-dep-major-update-SNAPSHOT/gravitee-bom-4.0.0-dep-major-update-SNAPSHOT.zip)

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -60,25 +60,25 @@
     </organization>
 
     <properties>
-        <assertj-core.version>3.22.0</assertj-core.version>
-        <jackson.version>2.13.2.1</jackson.version>
+        <assertj-core.version>3.24.2</assertj-core.version>
+        <jackson.version>2.14.0</jackson.version>
```
Comment (gaetanmaisse) [inline pom.xml:64]: Should we move to 2.14.2 directly @jhaeyaert @guillaumelamirand ? For now, I used the version defined in the vertx-bom

Comment (guillaumelamirand) [reply pom.xml:64]: I would say go with 2.14.2, most of the time patch version are security fixes. I let Jeoffrey gives his opinion.

Comment (jhaeyaert) [reply pom.xml:64]: Agree

**Review by jhaeyaert — APPROVED** [review]

**Pull request merged**